client.go

```diff
@@ -520,6 +526,269 @@ func (c *Client) ListImages(ctx context.Context, filters ...string) ([]Image, er
 	return images, nil
 }
 
+// Checkpoint creates a checkpoint of the container
+func (c *Client) Checkpoint(ctx context.Context, id, ref string, opts ...CheckpointOpts) error {
+	index := &ocispec.Index{
+		Versioned: ver.Versioned{
+			SchemaVersion: 2,
+		},
+		Annotations: make(map[string]string),
+	}
+	copts := &options.CheckpointOptions{
```

Comment:
These will have to be moved to an Opt

Reply:
These are passed to the checkpoint opt func to change anything they want.  Did you mean that or something else?

Reply:
We also may need to save this as part of the checkpoint because it is also used on restore

Reply:
ya that works